zeebe/protocol-impl/src/main/java/io/camunda/zeebe/protocol/impl/record/value/usertask/UserTaskRecord.java

```diff
@@ -103,8 +103,42 @@ public final class UserTaskRecord extends UnifiedRecordValue implements UserTask
   private final LongProperty elementInstanceKeyProp = new LongProperty("elementInstanceKey", -1L);
   private final StringProperty tenantIdProp =
       new StringProperty("tenantId", TenantOwned.DEFAULT_TENANT_IDENTIFIER);
+
+  /**
+   * Tracks the names of user task attributes that have been modified.
+   *
+   * <p>This property was introduced to enable tracking of attributes that were truly modified.
+   * Since our `msgpack` implementation doesn't support `null` values, all record attributes have
+   * default non-null values. As a result, without `changedAttributesProp`, it would be impossible
+   * to determine whether an attribute was explicitly modified or simply retained its default value.
+   * This property solves that issue by explicitly listing attributes that were changed.
```

Comment:
🔧 We can add some more clarity to the first sentence by being a bit more explicit. This allows us to remove the second sentence. I also suggest to use `changedAttributes` over `changedAttributesProp` as this is the name of the property.

```suggestion
   * Tracks the names of user task attributes that are intended to be modified (e.g. on `UPDATE`), or have been modified (e.g. on `UPDATED`).
   *
   * Since our `msgpack` implementation doesn't support `null` values, all record attributes have
   * default non-null values. As a result, without `changedAttributes`, it would be impossible
   * to determine whether an attribute was explicitly modified or simply retained its default value.
   * This property solves that issue by explicitly listing attributes that were changed.
```